File: tests/snapshots/iterators.rs
// Iterators — filter, map, sum, any, count, collect

fn main() {
    let mut nums: Vec<i64> = Vec::new();
    nums.push(1);
    nums.push(2);
    nums.push(3);
    nums.push(4);
    nums.push(5);
    let big: Vec<i64> = nums.iter().copied().filter(|&x| x > 2).collect();
    let doubled: Vec<i64> = nums.iter().copied().map(|x| x * 2).collect();
    let total: i64 = nums.iter().copied().sum();
    let has_big: bool = nums.iter().copied().any(|x| x > 4);
    println!("count:   {}", nums.iter().copied().count());
    println!("total:   {}", total);
    println!("has big: {}", has_big);
    for n in &big {
        println!("big:     {}", *n);
    }
    for n in &doubled {
        println!("doubled: {}", *n);
    }
}

File: tests/snapshots/iterators.py
# Iterators — filter, map, sum, any, count, collect

def _vb(x):
    if isinstance(x, bool):
        return "true" if x else "false"
    if isinstance(x, float) and x.is_integer():
        return str(int(x))
    return str(x)

def main():
    nums: list[int] = []
    nums.append(1)
    nums.append(2)
    nums.append(3)
    nums.append(4)
    nums.append(5)
    big: list[int] = [x for x in nums if x > 2]
    doubled: list[int] = [x * 2 for x in nums]
    total: int = sum(nums)
    hasbig: bool = any(x > 4 for x in nums)
    print(f"count:   {_vb(len(nums))}")
    print(f"total:   {_vb(total)}")
    print(f"has big: {_vb(hasbig)}")
    for n in big:
        print(f"big:     {_vb(n)}")
    for n in doubled:
        print(f"doubled: {_vb(n)}")


if __name__ == "__main__":
    main()
